### PR TITLE
aamitra/vz-4216- Support for private dns scope 

### PR DESCRIPTION
**Pull request by theabmitra**
# Description

Please include a summary of the change and which issue is fixed.  If there are any dependencies, for example on a PR in another repository, please list those.

Fixes VZ-9999

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: platform-operator/verrazzano-bom.json [hunk 1]
```diff
@@ -83,8 +83,8 @@
           "name": "external-dns",
           "images": [
             {
-              "image": "external-dns",
-              "tag": "v0.7.1-20211125051406-fb54e02b",
+              "image": "external-dns-jenkins",
```
[the remark this answers is not in the document]
Comment (theabmitra) [reply platform-operator/verrazzano-bom.json]: updated this with correct build from external-dns

**Review by michael-cico — DISMISSED** [review]

**Review by michael-cico — APPROVED** [review]

File: platform-operator/controllers/verrazzano/component/externaldns/external_dns.go [hunk 2]
```diff
@@ -42,6 +46,14 @@ func preInstall(compContext spi.ComponentContext) error {
 		return err
 	}
 
+	//check if scope value is valid
+	scope := dns.OCI.DNSScope
+	if strings.ToLower(scope) != dnsGlobal && strings.ToLower(scope) != dnsPrivate && scope != "" {
+		message := fmt.Sprintf("Invalid oci dns scope value: %s. If set ,value can only be 'Global' or 'Private", dns.OCI.DNSScope)
```
Comment (markxnelson) [inline platform-operator/controllers/verrazzano/component/externaldns/external_dns.go:52]: OCI is product name, must be capitalized.
DNS needs to be capitalized.
Space after comma, not before.
Missing closing quote

File: platform-operator/controllers/verrazzano/component/externaldns/external_dns.go [hunk 3]
```diff
@@ -42,6 +46,14 @@ func preInstall(compContext spi.ComponentContext) error {
...
+		compContext.Log().Errorf(message)
```
Comment (markxnelson) [inline platform-operator/controllers/verrazzano/component/externaldns/external_dns.go:53]: errorf but no format string

Comment (theabmitra) [reply platform-operator/controllers/verrazzano/component/externaldns/external_dns.go:53]: the message is already formatted with Sprintf

**Review by michael-cico — APPROVED** [review]

File: platform-operator/controllers/verrazzano/component/externaldns/external_dns.go (hunk 3 above)
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/externaldns/external_dns.go:53]: I think what Mark means is you can perform the formatting and params in the call to Errorf and skip the fmt.Sprintf, but it works.

**Pull request merged**